src/daemon/ca/certauth.rs

```diff
@@ -207,113 +232,168 @@ impl Aggregate for CertAuth {
+            }
 
-            EvtDet::ChildUpdatedResources(child, resources) => {
+            CaEvtDet::ChildUpdatedResources { child, resources } => {
                 self.children.get_mut(&child).unwrap().set_resources(resources)
             }
 
-            EvtDet::ChildRemoved(child) => {
+            CaEvtDet::ChildRemoved { child } => {
                 self.children.remove(&child);
             }
 
             //-----------------------------------------------------------------------
             // Being a child
             //-----------------------------------------------------------------------
-            EvtDet::IdUpdated(id) => {
+            CaEvtDet::IdUpdated { id } => {
                 self.id = id;
             }
-            EvtDet::ParentAdded(handle, info) => {
-                self.parents.insert(handle, info);
+            CaEvtDet::ParentAdded { parent, contact } => {
+                self.parents.insert(parent, contact);
             }
-            EvtDet::ParentUpdated(handle, info) => {
-                self.parents.insert(handle, info);
+            CaEvtDet::ParentUpdated { parent, contact } => {
+                self.parents.insert(parent, contact);
             }
-            EvtDet::ParentRemoved(handle, _deltas) => {
-                self.parents.remove(&handle);
-                self.resources.retain(|_, rc| rc.parent_handle() != &handle);
+            CaEvtDet::ParentRemoved { parent } => {
+                self.parents.remove(&parent);
+                self.resources.retain(|_, rc| rc.parent_handle() != &parent);
             }
 
-            EvtDet::ResourceClassAdded(name, rc) => {
+            CaEvtDet::ResourceClassAdded {
+                resource_class_name,
+                parent,
+                parent_resource_class_name,
+                pending_key,
+            } => {
                 self.next_class_name += 1;
-                self.resources.insert(name, rc);
+                let ns = resource_class_name.to_string();
+                let rc = ResourceClass::create(
+                    resource_class_name.clone(),
+                    ns,
+                    parent,
+                    parent_resource_class_name,
+                    pending_key,
+                );
+                self.resources.insert(resource_class_name, rc);
             }
-            EvtDet::ResourceClassRemoved(name, _delta, _parent, _revocations) => {
-                self.resources.remove(&name);
+            CaEvtDet::ResourceClassRemoved {
+                resource_class_name, ..
+            } => {
+                self.resources.remove(&resource_class_name);
             }
-            EvtDet::CertificateRequested(name, req, status) => {
-                self.resources.get_mut(&name).unwrap().add_request(status, req);
+            CaEvtDet::CertificateRequested {
+                resource_class_name,
+                req,
+                ki,
```

Comment:
I find the former `key_id` more self-evident, and in many places you have used longer more explicit names now. Was this change to `ki` intentional here?